src/screens/map-screen/map-screen.component.tsx

```diff
@@ -144,6 +143,7 @@ const MapScreen = () => {
         Buildings.forEach((building) => {
           if (isPointInPolygon(response.coords, building.boundingBox)) {
             onBuildingTap(building);
+            setStartLocationDisplay(building.displayName);
```

Comment:
FIX-NOTE-1 (Please refer to the comment above mentionning this one)

Please don't set just the display name, set the actual startLocation as such: 
`setStartLocation(buildingToMarker(building));`

Don't forget to rebase too before doing these changes.